tools/read_insitu_diagnostics.py

```diff
@@ -57,23 +58,74 @@ def emittance_y(all_data):
                   * (all_data["[uy^2]"] - all_data["[uy]"]**2) \
                   - (all_data["[y*uy]"] - all_data["[y]"]*all_data["[uy]"])**2))
 
-def energy_spread(all_data):
-    return np.sqrt(np.maximum(all_data["[ga^2]"]-all_data["[ga]"]**2,0))
+def gamma_mean(all_data):
+    return all_data["[ga]"]
+
+def energy_mean_eV(all_data, per_slice=False):
+    if per_slice:
+        if all_data["is_normalized_units"][0]:
+            return (constants.m_e * constants.c**2 / constants.e) * gamma_mean(all_data) * np.atleast_2d(all_data["mass"]).T
+        else:
+            return (constants.c**2 / constants.e) * gamma_mean(all_data) * np.atleast_2d(all_data["mass"]).T
+    else:
+        if all_data["is_normalized_units"][0]:
+            return (constants.m_e * constants.c**2 / constants.e) * gamma_mean(all_data["average"]) * all_data["mass"]
+        else:
+            return (constants.c**2 / constants.e) * gamma_mean(all_data["average"]) * all_data["mass"]
+
+def gamma_spread(all_data):
+    return np.sqrt(np.maximum(all_data["[ga^2]"] - all_data["[ga]"]**2,0))
+
+def energy_spread_eV(all_data, per_slice=False):
+    if per_slice:
+        if all_data["is_normalized_units"][0]:
+            return (constants.m_e * constants.c**2 / constants.e) * gamma_spread(all_data) * np.atleast_2d(all_data["mass"]).T
+        else:
+            return (constants.c**2 / constants.e) * gamma_spread(all_data) * np.atleast_2d(all_data["mass"]).T
+    else:
+        if all_data["is_normalized_units"][0]:
+            return (constants.m_e * constants.c**2 / constants.e) * gamma_spread(all_data["average"]) * all_data["mass"]
+        else:
+            return (constants.c**2 / constants.e) * gamma_spread(all_data["average"]) * all_data["mass"]
 
 def position_mean_x(all_data):
     return all_data["[x]"]
 
 def position_mean_y(all_data):
     return all_data["[y]"]
 
+def position_mean_z(all_data):
+    return all_data["[z]"]
+
 def position_std_x(all_data):
     return np.sqrt(np.maximum(all_data["[x^2]"] - all_data["[x]"]**2,0))
 
 def position_std_y(all_data):
     return np.sqrt(np.maximum(all_data["[y^2]"] - all_data["[y]"]**2,0))
 
+def position_std_z(all_data):
+    return np.sqrt(np.maximum(all_data["[z^2]"] - all_data["[z]"]**2,0))
+
+def normalized_momentum_mean_x(all_data):
+    return all_data["[ux]"]
+
+def normalized_momentum_mean_y(all_data):
+    return all_data["[uy]"]
+
+def normalized_momentum_mean_z(all_data):
+    return all_data["[uz]"]
+
```

Comment:
Such direct accessors may bring more confusion than they help, the user should probably just be familiar with `all_data`. I think we should remove them (that include `position_mean_x/y/z`). All other functions are less straightforward and should be kept. What do you think?